pkg/resource/plugin/analyzer.go

```diff
@@ -31,7 +32,7 @@ type Analyzer interface {
 	// Name fetches an analyzer's qualified name.
 	Name() tokens.QName
 	// Analyze analyzes a single resource object, and returns any errors that it finds.
-	Analyze(t tokens.Type, props resource.PropertyMap) ([]AnalyzeFailure, error)
+	Analyze(t tokens.Type, props resource.PropertyMap) ([]*pulumirpc.AnalyzeDiagnostic, error)
```

Comment:
FWIW, in general we have tried to keep from leaking gRPC types across these higher level interfaces (in favor of having a go type like `AnalyzeFailure` and then the provider that manages analyzers from plugins which implements this interface is responsible for translating the rpc types up as part of implementing the method.